.github/workflows/vulnerability_scan_subworkflow.yml

```diff
@@ -1,78 +1,33 @@
 name: Vulnerability Scan
 
 on:
-    workflow_call:
-        inputs:
-            ref:
-                required: true
-                type: string
-        secrets:
-            SNYK_TOKEN:
-                required: true
-    workflow_dispatch:
-        inputs:
-            ref:
-                required: true
-                type: string
+  workflow_call:
+    inputs:
+      ref:
+        required: true
+        type: string
+    secrets:
+      SNYK_TOKEN:
+        required: true
+  workflow_dispatch:
+    inputs:
+      ref:
+        required: true
+        type: string
 
 jobs:
-  scan-oss:
-    env:
-      DOCKLE_HOST: "unix:///var/run/docker.sock"
+  scan:
     runs-on: ubuntu-latest
-    steps:
-      - name: Checkout Code at ${{ inputs.ref }} branch
-        uses: actions/checkout@v4
-        with:
-            ref: ${{ inputs.ref }}
-
-      - name: Install xmllint
-        uses: ./.github/actions/install-xmllint
-
-      - name: Get OSS dist ZIP
-        run: |
-          . .github/scripts/oss-build.functions.sh
-          HZ_VERSION=$(awk -F '=' '/^ARG HZ_VERSION=/ {print $2}' hazelcast-oss/Dockerfile)
-          export HZ_SNAPSHOT_INTERNAL_PASSWORD=${{ secrets.HZ_SNAPSHOT_INTERNAL_PASSWORD }}
-          export HZ_SNAPSHOT_INTERNAL_USERNAME=${{ secrets.HZ_SNAPSHOT_INTERNAL_USERNAME }}
-          HAZELCAST_OSS_ZIP_URL=$(get_hz_dist_zip "" "${HZ_VERSION}")        
-          curl --fail --silent --show-error --location "$HAZELCAST_OSS_ZIP_URL" --output hazelcast-oss/hazelcast-distribution.zip;
-
-      - name: Build OSS image
-        run: |
-            docker build -t hazelcast/oss:${{ github.sha }} hazelcast-oss
-
-      - name: Scan OSS image by Trivy
-        if: always()
-        uses: aquasecurity/trivy-action@0.23.0
-        with:
-          image-ref: hazelcast/oss:${{ github.sha }}
-          trivy-config: .github/containerscan/trivy.yaml
-
-      - name: Scan OSS image by Dockle
-        if: always()
-        uses: goodwithtech/dockle-action@main
-        with:
-          image: hazelcast/oss:${{ github.sha }}
-          format: 'list'
-          exit-code: '1'
-          exit-level: 'warn'
-          # too many false positives, we don't use credentials in Dockerfile
-          ignore: 'CIS-DI-0010'
-
-      - name: Scan OSS image by Snyk
-        if: always()
-        uses: snyk/actions/docker@master
-        env:
-          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
-        with:
-          image: hazelcast/oss:${{ github.sha }}
-          args: --file=hazelcast-oss/Dockerfile --policy-path=.github/containerscan --severity-threshold=high --exclude-base-image-vulns
-
-  scan-ee:
+    strategy:
+      matrix:
+        image:
+          - label: oss
+            distribution_zip_name: hazelcast-distribution.zip
+          - label: enterprise
+            distribution_zip_name: hazelcast-enterprise-distribution.zip
     env:
-      DOCKLE_HOST: "unix:///var/run/docker.sock"
```

Comment:
Not need anymore?

Reply:
> Not need anymore?

I've moved it to the step it's actually required for:

https://github.com/hazelcast/hazelcast-docker/blob/5978b345bb8f1911e8294b327eff7cc99fee4b7a/.github/workflows/vulnerability_scan_subworkflow.yml#L61-63